### PR TITLE
Generate dispatch code from C source

### LINKED ISSUE #2816: Generate built-in dispatch.ll from C source
It looks like there is no real reason to have dispatch*.ll files manually generated from `dispatch.c` source. It can be generated during ISPC build process.

Due to missing tests for dispatch logic, it depends on #2815

### PR DESCRIPTION
**Pull request by nurmukhametov**
This PR fixes #2816

### CODE REVIEW

**Review by aneshlya — APPROVED** [review]
Looks good!

File: .github/workflows/ispc-ci.yml [hunk 1]
```diff
@@ -75,6 +75,55 @@ jobs:
         $RUN_SMOKE && echo "optsets=${OPTSETS_SMOKE}" >> $GITHUB_OUTPUT || true
         $RUN_FULL &&  echo "optsets=${OPTSETS_FULL}" >> $GITHUB_OUTPUT || true
 
+  linux-aarch64-build-ispc:
```
[the remark this answers is not in the document]
Comment (nurmukhametov) [reply .github/workflows/ispc-ci.yml:78]: Cross compiler job issue #2881 .

I hope that we will have public runners with arm64 soon.  https://github.blog/2024-06-03-arm64-on-github-actions-powering-faster-more-efficient-build-systems/

[the remark this answers is not in the document]
Comment (dbabokin) [reply .github/workflows/ispc-ci.yml:78]: Thank for the link, I missed that news.

Anyway, using self-hosted for now is more than good enough.

And probably for rebuilding clang we might need to stay on self-hosted, as it's faster.

File: .github/workflows/ispc-ci.yml [hunk 2]
```diff
@@ -540,8 +541,13 @@ jobs:
         tar xf $LLVM_TAR
         echo "${GITHUB_WORKSPACE}/bin-$LLVM_VERSION/bin" >> $GITHUB_PATH
         brew install bison flex
-        echo "/usr/local/opt/bison/bin" >> $GITHUB_PATH
-        echo "/usr/local/opt/flex/bin" >> $GITHUB_PATH
+        if [ "${{ matrix.runner }}" == "macos-14" ]; then
```
[the remark this answers is not in the document]
Comment (dbabokin) [reply .github/workflows/ispc-ci.yml:593]: Ah! Now I got it, makes sense.

**Pull request merged**